concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/TreeStateTest.hs

```diff
@@ -0,0 +1,908 @@
+    tr2 = dummyTransactionWithPayload 1 (EncodedPayload "a")
+    cu0 = dummyUpdateInstructionWM 1
+    cu1 = dummyUpdateInstructionWM 2
+    cred0 = credentialDeploymentWM
+    addTrans t = snd . addTransaction (normalTransaction t) 0 (dummySuccessTransactionResult (transactionNonce t))
+    addChainUpdate u = snd . addTransaction (chainUpdate u) 0 (dummySuccessTransactionResult (updateSeqNumber $ uiHeader $ wmdData u))
+    addCredential = snd . addTransaction dummyCredentialDeployment 0 dummySuccessCredentialDeployment
+    credDeploymentHash = getHash dummyCredentialDeployment
+    sd =
+        dummyInitialSkovData
+            & transactionTable
+                %~ addTrans tr0
+                . addTrans tr1
+                . addTrans tr2
+    sd1 =
+        dummyInitialSkovData
+            & transactionTable
+                %~ addChainUpdate cu0
+                . addChainUpdate cu1
+                . addTrans tr1
+    sd2 =
+        dummyInitialSkovData
+            & transactionTable
+                %~ addCredential
+                . addTrans tr1
+
+testPutTransaction :: Spec
+testPutTransaction = describe "putTransaction" $ do
+    it "put transaction" $ do
+        sd' <- execStateT (putTransaction tr0Round (normalTransaction tr0) (dummySuccessTransactionResult 1)) dummyInitialSkovData
+        assertEqual
+            "Account non-finalized transactions"
+            (Just AccountNonFinalizedTransactions{_anftNextNonce = 1, _anftMap = Map.singleton 1 (Map.singleton tr0 (dummySuccessTransactionResult 1))})
+            (sd' ^. transactionTable . ttNonFinalizedTransactions . at sender)
+        assertEqual
+            "transaction hash map"
+            (HM.fromList [(getHash tr0, (normalTransaction tr0, Received (commitPoint tr0Round) (dummySuccessTransactionResult 1)))])
+            (sd' ^. transactionTable . ttHashMap)
+        assertEqual
+            "transaction table purge counter is incremented"
+            (1 + dummyInitialSkovData ^. transactionTablePurgeCounter)
+            (sd' ^. transactionTablePurgeCounter)
+        sd'' <- execStateT (removeTransactions [normalTransaction tr0]) sd'
+        added <- evalStateT (putTransaction tr0Round (normalTransaction tr0) (dummySuccessTransactionResult 1)) sd''
+        assertEqual "tx should not be added" False added
+  where
+    tr0Round = 1
+    tr0 = dummyTransaction 1
+    sender = accountAddressEmbed dummyAccountAddress
+
+testCommitTransaction :: Spec
+testCommitTransaction = describe "commitTransaction" $ do
+    it "commit transaction" $ do
+        sd' <- execStateT (commitTransaction 1 bh 0 (normalTransaction tr0)) sd
+        assertEqual
+            "transaction hash map"
+            (HM.fromList [(getHash tr0, (normalTransaction tr0, Committed 1 (dummySuccessTransactionResult (transactionNonce tr0)) $ HM.fromList [(bh, TransactionIndex 0)]))])
+            (sd' ^. transactionTable . ttHashMap)
+  where
+    tr0 = dummyTransaction 1
```

Comment:
Plz 0 abbr

Reply:
transaction 0. I think it's ok?

Reply:
It's especially nice that you use `tr` whereas we most often use `tx` for transactions.